padre_meddea/io/amptek.py

```diff
@@ -0,0 +1,159 @@
+        for line in fp:
+            #  figure out what section of the file we are in
+            # the order used here is the order expected in the file
+            # once in new section you are done with the previous section
+            if line.count("PMCA SPECTRUM"):
+                in_spectrum_meta_section = True
+                continue
+            if line.count("CALIBRATION"):
+                in_calib_section = True
+                in_spectrum_meta_section = False
+                continue
+            if line.count("ROI"):
+                in_roi_section = True
+                in_calib_section = False
+                in_spectrum_meta_section = False
+                continue
+            if line.count("DATA"):
+                in_data_section = True
+                in_roi_section = False
+                continue
+            if line.count("END"):
+                in_data_section = False
+            if line.count("DP5 CONFIGURATION"):
+                in_dp5config_section = True
+                in_data_section = False
+                continue
+            if line.count("DPP STATUS"):
+                in_dp5config_section = False
+                in_status_section = True
+                continue
+
+            # now gather information
+            if in_spectrum_meta_section:  # this is a measurement metadata tag
+                label = line.split("-")[0].strip()
+                value = line.split("-")[1].strip()
+                try:
+                    value = float(value)
+                except ValueError:
+                    pass
+                hdr.append((label, value))
+            if in_calib_section:
+                if line.count("-"):
+                    continue
+                else:
+                    channel = float(line.split(" ")[0])
+                    energy = float(line.split(" ")[1])
+                    calib_data.update({channel: energy})
+            if in_roi_section:
+                roi_data.update({int(line.split(" ")[0]): int(line.split(" ")[1])})
+            if in_data_section and not (line.count("<<") == 1):
+                data.append(int(line))
+            if in_dp5config_section and not (line.count("<<") == 1):
+                keyword = line.split("=")[0].upper()
+                value = line.split("=")[1].split(";")[0].strip()
+                description = line.split("=")[1].split(";")[1].strip()
+                try:
+                    value = float(value)
+                except ValueError:
+                    pass
+                hdr.append((keyword.replace(" ", "")[0:7], value, description))
+            if in_status_section and not (line.count("<<") == 1):
+                keyword = line.split(":")[0].strip().upper()
+                value = line.split(":")[1].strip()
+                try:
+                    value = float(value)
+                except ValueError:
+                    pass
+                hdr.append((keyword.replace(" ", "")[0:7], value))
```

Comment:
Running Line 119 returns the following error: 
ERROR: ValueError: FITS header values must contain standard printable ASCII characters; '27°C' contains characters not representable in ASCII or non-printable characters. [astropy.io.fits.card]

I have not found a way to easily get around this. Perhaps the easiest thing to do is to modify the .mca file itself, since it's just this one character that is causing the problem.

Reply:
That's weird. My test mca file does not have this symbol in it (see [here](https://github.com/PADRESat/padre_meddea/blob/b1bd4b9ddaec478c612e148aee62a9980bc449bf/padre_meddea/data/test/minix_20kV_15uA_sdd.mca)) which has the lines 
2143 `TEC Temp: 252K`
2144 `Board Temp: 29C`
at the bottom of the file. Can you paste the offending line from your file here please?